### DISCUSSION TITLE
winGRASS and R integration/coupling

### DISCUSSION THREAD

**Discussion by hellik** [OSGeo/grass#3091] [Ideas]
for a long time now, winGRASS-R-integration is based upon [Rbatch-files](https://github.com/OSGeo/grass/tree/main/mswindows/external/rbatch).

winGRASS is based upon OSGeo4W. 

in **OSGeo4W version 1**, these files have been included as an own package, see https://download.osgeo.org/osgeo4w/v1/x86_64/release/rbatch/ 

there was a switch to OSGeo4W version 2, though these **Rbatch-files** have never been migrated to this OSGeo4W version (see [discussion](https://lists.osgeo.org/pipermail/grass-user/2023-July/083357.html))

these files are also living in the [GRASS source](https://github.com/OSGeo/grass/tree/main/mswindows/external/rbatch).

now we are in the situation, that winGRASS standalone installer provides this R integration/coupling, though OSgeo4W version 2 winGRASS does not.

there was a feature request to add RStudio to %PATH% on demand (see [winGRASS: add functionality for manually adding RStudio path to %PATH% within a GRASS session](https://github.com/OSGeo/grass/issues/1171). 

should we add such an on-demand-%PATH%-expansion also for R itsef, in case **Rbatch-files** functionality is not available.

any thoughts? @landam @ninsbl @neteler

**Comment by jef-n** [comment]
why are the files not included in the mswindows/osgeo4w/package.sh?

**Comment by hellik** [comment]
> why are the files not included in the mswindows/osgeo4w/package.sh?

Has to be reviewed

**Comment by jef-n** [comment]
> > why are the files not included in the mswindows/osgeo4w/package.sh?
> 
> Has to be reviewed

See also https://github.com/jef-n/OSGeo4W/blob/master/src/grass-dev/osgeo4w/patch#L180
